### PR TITLE
Adds Create separate CSS files per component section

### PR DESCRIPTION
**Pull request by thejimbirch**
**Summary**

Adds a section that instructs on how to create separate CSS files per component

Per discussion in https://emulsify.slack.com/archives/C4605SM9V/p1635530239020600

### CODE REVIEW

File: details/webpack-and-build.md [hunk 1]
```diff
@@ -18,6 +18,45 @@ The Webpack build for the project found in the `/webpack` directory can be exten
...
+* webpack
+* * css
+* * * main-css.js
+* * * custom-css.js
+
+3. In webpack.common.js replace `entries.css` with
+
+```
+// CSS Files.
+glob.sync(`${webpackDir}/css/*js`).forEach((file) => {
```
Comment (ModulesUnraveled) [inline details/webpack-and-build.md:51]: I'm getting an error that `ReferenceError: webpackDir is not defined.` Is there a line missing here?

**Review by ModulesUnraveled — CHANGES_REQUESTED** [review]

File: details/webpack-and-build.md (hunk 1 above)
Comment (ModulesUnraveled) [reply details/webpack-and-build.md:51]: I wonder if there's a difference in what version of Emulsify you are using, and the latest in `master`

Comment (asipple1) [reply details/webpack-and-build.md:51]: On line 6 I have the following for webpackDir
`const webpackDir = path.resolve(__dirname);`

File: details/webpack-and-build.md [hunk 2]
```diff
@@ -18,6 +18,45 @@ The Webpack build for the project found in the `/webpack` directory can be exten
...
+  filename: 'style.css',
+  chunkFilename: '[id].css',
+});
+
+// To this
+const MiniCssExtractPlugin = new _MiniCssExtractPlugin({
+  filename: "[name].css",
+  chunkFilename: "[id].css"
+});
+```
```
Comment (ModulesUnraveled) [inline details/webpack-and-build.md:38]:
```suggestion

1. In `webpack/plugins.js` change

```
const MiniCssExtractPlugin = new _MiniCssExtractPlugin({
  filename: 'style.css',
  chunkFilename: '[id].css',
});
```

To:

```
const MiniCssExtractPlugin = new _MiniCssExtractPlugin({
  filename: '[name].css',
  chunkFilename: '[id].css'
});
```
```

File: details/webpack-and-build.md [hunk 3]
```diff
@@ -18,6 +18,45 @@ The Webpack build for the project found in the `/webpack` directory can be exten
...
+
+2. In the webpack directory create a css directory with the following structure. Each .js imports their own .scss file.
+
+* webpack
+* * css
+* * * main-css.js
+* * * custom-css.js
```
Comment (ModulesUnraveled) [inline details/webpack-and-build.md:45]:
```suggestion
2. In the webpack directory create a css directory with the following structure. (Each js file should import its own `<component>.scss` file. Look at `webpack/css.js` for an example.)

```
.
└──webpack
│  └──css
│     │   component-1.js
│     │   component-2.js
```

```

File: details/webpack-and-build.md [hunk 4]
```diff
@@ -18,6 +18,45 @@ The Webpack build for the project found in the `/webpack` directory can be exten
...
+3. In webpack.common.js replace `entries.css` with
+
+```
+// CSS Files.
+glob.sync(`${webpackDir}/css/*js`).forEach((file) => {
+  const baseFileName = path.basename(file);
+  const newfilePath = `css/${baseFileName.replace('.js', '')}`;
+  entries[newfilePath] = file;
+});
+```
```
Comment (ModulesUnraveled) [inline details/webpack-and-build.md:56]:
```suggestion
3. In webpack.common.js replace

```
entries.css = path.resolve(webpackDir, 'css.js');
```

with

```
// CSS Files.
glob.sync(`${webpackDir}/css/*js`).forEach((file) => {
  const baseFileName = path.basename(file);
  const newfilePath = `css/${baseFileName.replace('.js', '')}`;
  entries[newfilePath] = file;
});
```
```

File: details/webpack-and-build.md [hunk 5]
```diff
@@ -18,6 +18,45 @@ The Webpack build for the project found in the `/webpack` directory can be exten
...
+
+4. Now you can create your own Drupal libraries in the Drupal theme to import the files where you need them.
```
Comment (ModulesUnraveled) [inline details/webpack-and-build.md:58]:
```suggestion
4. Now you can define CSS files in your Drupal libraries only where you need them, just like you do with js.

5. Finally, Storybook only loads the main `styles.scss` file, by default. So make sure you add anything not globbed into that file to `.storybook/preview.js`.
```

**Review by ModulesUnraveled — CHANGES_REQUESTED** [review]

**Review by ModulesUnraveled — APPROVED** [review]
Got the suggestions fixed!

**Pull request merged**